test/xbgpu/test_engine.py

```diff
@@ -487,64 +582,75 @@ def heap_factory(batch_index: int) -> list[spead2.send.HeapReference]:
             # Generate one extra chunk to simulate an incomplete accumulation
             # to check that dumps are aligned correctly - even if the first
             # received batch is from the middle of an accumulation.
-            batch_start_index = (
-                first_accumulation_index + 1
-            ) * heap_accumulation_threshold - HEAPS_PER_FENGINE_PER_CHUNK
-            batch_end_index = (first_accumulation_index + 1 + n_full_accumulations) * heap_accumulation_threshold
-            device_results = await self._send_data(
-                mock_recv_stream,
-                recv_stream,
-                heap_accumulation_threshold=heap_accumulation_threshold,
+            batch_start_index = (first_batch_index) * max_heap_acc_threshold - HEAPS_PER_FENGINE_PER_CHUNK
+            batch_end_index = (first_batch_index + min_full_accumulations) * max_heap_acc_threshold
```

Comment:
Should probably be
```suggestion
            batch_end_index = first_batch_index + min_full_accumulations * max_heap_acc_threshold
```
Also since `min_full_accumulations * max_heap_acc_threshold` is (currently) just the LCM, this could probably be simplified.

Reply:
As always, fair points. However I'm gona plead the Fifth and cite [James's comment above](https://github.com/ska-sa/katgpucbf/pull/600#discussion_r1296857624) to make this part of a technical-debt-type overhaul.

Your call is still important to us and we thank you for your patience.

Reply:
When you create a ticket Amish please just link to these comments. It'll help with the context.